File: pset1/mario/less/mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // initialise integers
    int n = 0;
    int outerLoop = 0;
    int innerLoop = 0;
    do
    {
        // get pyramid height from user
        n = get_int("Number please between 0 - 23: ");
    }

    // validate user input
    while (n < 0 || n > 23);

    // outer loop
    for (outerLoop = 0; outerLoop < n; outerLoop++)
    {
        // inner loop
        for (innerLoop = 0; innerLoop < n + 1; innerLoop++)
        {
            // algorythm for creating pyramid shape
            if (innerLoop + outerLoop < n - 1)
        {
            // print spaces
            printf(" ");
        }
            else
        {
            // print hash tags
            printf("#");
        }

        }
        // start new line
        printf("\n");
    }
}

File: pset6/mario/less/mario.py
from cs50 import get_int

while True:
    # get size of pyramid from the user
    n = get_int("Number please between 0 - 23: ")
    if n >= 0 and n <= 23:
        break

for i in range(n):
# c outerLoop start

    for j in range(n + 1):
        # c innerLoop

        if i + j < n - 1:
            # algorithm for pyramid shape
            print(" ", end="")
        else:
            print("#", end="")

# c outerLoop end
    print()
